In this notebook are loaded only some data that could be useful for the preliminary analyses, in particular:
- **Clinical Data** from S1_Clinical_Annotation_Table.
- **RNA-Seq** data from Table_S12_RNA_Gene_Counts sheet contained in the excel file.
- **WES** data from a .txt file.

**Goal**: save the clinical annotation table, RNA-Seq data, and WES data only for the patients who are considered as (High Quality Samples, HQ).

### Import Utils and Setup

In [2]:
## Communication drive-colab
from google.colab import drive
import warnings
import os

## Data Structure and Data Analysis
import pandas as pd
import numpy as np

## Read rds files
!pip install pyreadr
import pyreadr

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 434.8/434.8 kB 4.5 MB/s eta 0:00:00


In [3]:
## Mount drive
drive.mount("/content/drive", force_remount = True)

Mounted at /content/drive


In [4]:
## Toggle as needed
warnings.filterwarnings("ignore")

## To see the maximum number of columns
pd.set_option("display.max_columns", None)

## Source path
source_path_data = "/content/drive/MyDrive/Tesi/Code/Source Data/"

## Save path
save_path_data     = "/content/drive/MyDrive/Tesi/Code/Personal_Code/Saved/Data/"
save_path_features = "/content/drive/MyDrive/Tesi/Code/Personal_Code/Saved/Features/"

## Import different type of data from their respective folders
source_path_data_clinical = source_path_data + "Clinical/"
source_path_data_rna      = source_path_data + "RNA/"
source_path_data_exome    = source_path_data + "Exome/"
source_path_data_ref      = source_path_data + "Reference/"

### Read Clinical Data

In [5]:
## Load Annotation
master_annotations_df = pd.read_excel(source_path_data_clinical + "/Table_S1_Clinical_Annotations.xlsx", sheet_name = "Table_S1_Clinical_Annotations", skiprows = 2)

In [6]:
## Check
print("Shape is: ", master_annotations_df.shape)
display(master_annotations_df.head())

Shape is:  (393, 38)


WES_Cohort_1  WES_Cohort_2  WES_All  RNA_Cohort_1  RNA_Cohort_2  RNA_All  \
0           1.0           NaN      1.0           NaN           NaN      NaN   
1           NaN           NaN      NaN           1.0           NaN      1.0   
2           1.0           NaN      1.0           1.0           NaN      1.0   
3           1.0           NaN      1.0           1.0           NaN      1.0   
4           NaN           NaN      NaN           1.0           NaN      1.0   

        Institution Harmonized_SU2C_Participant_ID_v2  \
0  Cleveland Clinic                  SU2CLC-CLE-NIVO1   
1  Cleveland Clinic                 SU2CLC-CLE-NIVO10   
2  Cleveland Clinic                 SU2CLC-CLE-NIVO18   
3  Cleveland Clinic                 SU2CLC-CLE-NIVO19   
4  Cleveland Clinic                  SU2CLC-CLE-NIVO2   

  Harmonized_SU2C_WES_Tumor_Sample_ID_v2  \
0                    SU2CLC-CLE-NIVO1-T1   
1                                    NaN   
2                   SU2CLC-CLE-NIVO18-T1   
3                   SU2CLC-CLE-NIVO19-T1   
4                                    NaN   

  Harmonized_SU2C_WES_Normal_Sample_ID_v2  \
0                     SU2CLC-CLE-NIVO1-N1   
1                                     NaN   
2                    SU2CLC-CLE-NIVO18-N1   
3                    SU2CLC-CLE-NIVO19-N1   
4                                     NaN   

  Harmonized_SU2C_RNA_Tumor_Sample_ID_v2 Pre-treatment_RNA_Sample_QC  \
0                                    NaN                         NaN   
1                   SU2CLC-CLE-NIVO10-T1                        Flag   
2                   SU2CLC-CLE-NIVO18-T1                        Keep   
3                   SU2CLC-CLE-NIVO19-T1                        Keep   
4                    SU2CLC-CLE-NIVO2-T1                        Keep   

  Patient_Age_at_Diagnosis Patient_Sex  Patient_Race  Patient_Smoking_Status  \
0                       61           F           0.0                     1.0   
1                       55           F           2.0                     1.0   
2                       68           F           0.0                     0.0   
3                       57           F           2.0                     2.0   
4                       63           F           0.0                     0.0   

   Patient_Smoking_Pack_Years_Harmonized Histology_Harmonized  \
0                                   40.0             Squamous   
1                                   30.0                Adeno   
2                                    0.0                Adeno   
3                                   15.0                Adeno   
4                                    0.0                Adeno   

  Histology_Detail  Initial_Stage Initial_Stage_Substage  PDL1_TPS  \
0              NaN            3.0                      A       NaN   
1              NaN            4.0                    NaN       NaN   
2              NaN            2.0                      A       NaN   
3              NaN            4.0                    NaN       NaN   
4              NaN            4.0                    NaN       NaN   

  PDL1_TPS_Description Local_Antibody_Clone Clinical_Driver  \
0                  NaN                  NaN             NaN   
1                  NaN                  NaN             NaN   
2                  NaN                  NaN            EGFR   
3                  NaN                  NaN             NaN   
4                  NaN                  NaN             NaN   

  Sequencing_Platform  Advanced_Diagnosis_Date  Line_of_Therapy  Agent_PD1  \
0                 NaN                    -39.0              2.0  Nivolumab   
1                 NaN                   -321.0              3.0  Nivolumab   
2                 NaN                   -533.0              4.0  Nivolumab   
3                 NaN                    -35.0              1.0  Nivolumab   
4                 NaN                   -262.0              4.0  Nivolumab   

  Agent_PD1_Category  Prior_Platinum  Prior_TKI  Harmonized_PFS_Event  \
0 

In [7]:
## Save the whole dataset
with open(save_path_data + "master_annotation.csv", "w") as output:
    output.write(str(master_annotations_df.to_csv()))

Description of Features:

- **WES_COHORT_1**:	Exome sample in cohort 1, (n = 248)
- **WES_COHORT_2**:	Exome sample in cohort 2, (n = 61)
- **WES_ALL**:	Exome sample available, (n = 309)
- **RNA_COHORT_1**:	RNA-Seq sample in cohort 1, (n = 122)
- **RNA_COHORT_2**:	RNA-Seq sample in cohort 2, (n = 30)
- **RNA_ALL**:	RNA-Seq sample available, (n = 152)
- **INSTITUTION**:	Institution
- **HARMONIZED_SU2C_WES_TUMOR_SAMPLE_ID_V2**: Tumor exome Sample ID
- **HARMONIZED_SU2C_WES_NORMAL_SAMPLE_ID_V2**: Normal exome Sample ID
- **HARMONIZED_SU2C_RNA_TUMOR_SAMPLE_ID_V2**: Tumor RNA-Seq Sample ID
- **PRE-TREATMENT_RNA_SAMPLE_QC**: Tumor RNA-Seq sample quality
- **PATIENT_AGE_AT_DIAGNOSIS**:	Patient age at cancer diagnosis
- **PATIENT_SEX**: Patient sex
- **PATIENT_RACE**:	Patient race, ({0:**Caucasian**, 1:**Asian**, 2:**African Amerian**, 3:**Other**})
- **PATIENT_SMOKING_STATUS**: Patient smoking status, ({0:**Never**, 1:**Former**, 2:**Current**})
- **PATIENT_SMOKING_PACK_YEARS_HARMONIZED**: Patient smoking pack years
- **HISTOLOGY_HARMONIZED**:	Tumor histology
- **HISTOLOGY_DETAIL**:	Tumor histology detail
- **INITIAL_STAGE**: Initial tumor stage
- **INITIAL_STAGE_SUBSTAGE**: Initial tumor substage
- **PDL1_TPS**: Tumor PDL1 Tumor Proportion Score (TPS) value
- **PDL1_TPS_DESCRIPTION**:	Tumor PDL1 Tumor Proportion Score (TPS) description
- **LOCAL_ANTIBODY_CLONE**:	Local PDL1 antibody clone
- **CLINICAL_DRIVER**: Clinical driver identified by targeted sequencing
- **SEQUENCING_PLATFORM**: Targeted sequencing platform
- **ADVANCED_DIAGNOSIS_DATE**:	Date of advanced diagnosis relative to PD-(L)1 treatment start
- **LINE_OF_THERAPY**:	Line of therapy that PD-(L)1 agent was given
- **AGENT_PD1**: PD-(L)1 agent received
- **AGENT_PD1_CATEGORY**: PD-(L)1 agent category
- **PRIOR_PLATINUM**: Platinum treatment prior to PD-(L)1 agent
- **PRIOR_TKI**: Tyrosine kinase inhibitor (TKI) prior to PD-(L)1 agent
- **HARMONIZED_PFS_EVENT**: Progression-free survival (PFS) event, (0=**Censor**)
- **HARMONIZED_PFS_DAYS**: Progression-free survival (PFS) in days
- **HARMONIZED_CONFIRMED_BOR**: Confirmed best overall response (BOR), ({CR:**Complete Response**, PR:**Partial Response**, SD:**Stable Disease**, PD:**Progressive Disease**, NE:**Not Evaluable**})
- **HARMONIZED_BOR_RECIST**: RECISTv1.1 measurement corresponding to best overall response (BOR) (e.g., 1 = 100%)
- **HARMONIZED_OS_EVENT**: Overall survival (OS) event, (0:**Censor**)
- **HARMONIZED_OS_DAYS**: Overall survival (OS) in days

The features are collected by 4 different sources:
- **Analysis Cohort (n = 393)**: {WES_Cohort_1, WES_Cohort_2, WES_All,	RNA_Cohort_1, RNA_Cohort_2, RNA_All}

- **Sample Information**: {Institution, Harmonized_SU2C_Participant_ID_v2, Harmonized_SU2C_WES_Tumor_Sample_ID_v2, Harmonized_SU2C_WES_Normal_Sample_ID_v2, Harmonized_SU2C_RNA_Tumor_Sample_ID_v2,Pre-treatment_RNA_Sample_QC}

- **Clinical Annotation**: {Patient_Age_at_Diagnosis, Patient_Sex, Patient_Race, Patient_Smoking_Status, Patient_Smoking_Pack_Years_Harmonized, Histology_Harmonized, Histology_Detail, Initial_Stage, Initial_Stage_Substage, PDL1_TPS, PDL1_TPS_Description, Local_Antibody_Clone, Clinical_Driver, Sequencing_Platform, Advanced_Diagnosis_Date, Line_of_Therapy, Agent_PD1, Agent_PD1_Category, Prior_Platinum, Prior_TKI}

- **Response**: {Harmonized_PFS_Event, Harmonized_PFS_Days, Harmonized_Confirmed_BOR, Harmonized_BOR_RECIST, Harmonized_OS_Event, Harmonized_OS_Days}

This information is obtained from this project: dbGaP Study Accession [phs002822.v1.p1](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/GetListOfAllObjects.cgi?study_id=phs002822.v1.p1&object_type=dataset) and it includes 4 different datasets.

The dataset that is loaded above is: [pht012388.v1.p1, SU2C_MARK_Subject_Phenotypes](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/dataset.cgi?study_id=phs002822.v1.p1&pht=12388)

Other Dataets are:

- [pht012386.v1.p1, SU2C_MARK_Subject](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/dataset.cgi?study_id=phs002822.v1.p1&pht=12386) that contains 3 features {**ID**, **Consent**, **Sex**}
- [pht012387.v1.p1, SU2C_MARK_Sample](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/dataset.cgi?study_id=phs002822.v1.p1&phv=504453&phd=&pha=&pht=12387&phvf=&phdf=&phaf=&phtf=&dssp=1&consent=&temp=1) that contains 2 features {**ID**, **Sample ID**}
- [pht012389.v1.p1, SU2C_MARK_Subject_Phenotypes](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/dataset.cgi?study_id=phs002822.v1.p1&pht=12389) that contains 3 features {**Sample ID**, **Analyte Type**, **Is Tumor**}

The reference [paper](https://www.nature.com/articles/s41588-023-01355-5#data-availability) has selected 152 samples out of 309 that will be used for RNA analysis, based on the quality of the observed samples considering the median exon TPM (CV), the number of genes detected, and other measures.

### Read RNA Counts

Considering the Excel file **SU2C-MARK_Supplementary_Tables_Combined_v5_Filtered**, the data loaded in the following chunk are **RNA-Seq Gene Counts** for SU2C-MARK Cohort.

In [8]:
## Load rna counts
su2c_rna_counts_mat = pd.read_csv(source_path_data_rna + "SU2C-MARK_Harmonized_rnaseqc_counts_v1.gct", skiprows = 2, sep = "\t").set_index("Name")

In [9]:
## Check
print("Shape is: ", su2c_rna_counts_mat.shape)
display(su2c_rna_counts_mat.head())

Shape is:  (57523, 153)


Description  SU2CLC-CLE-NIVO10-T1  SU2CLC-CLE-NIVO18-T1  \
Name                                                                         
ENSG00000223972.5      DDX11L1                     0                     0   
ENSG00000227232.5       WASH7P                    95                    96   
ENSG00000278267.1    MIR6859-1                     0                     0   
ENSG00000243485.5  MIR1302-2HG                     0                     0   
ENSG00000284332.1    MIR1302-2                     0                     0   

                   SU2CLC-CLE-NIVO19-T1  SU2CLC-CLE-NIVO2-T1  \
Name                                                           
ENSG00000223972.5                    23                   36   
ENSG00000227232.5                   663                  395   
ENSG00000278267.1                     0                    0   
ENSG00000243485.5                     0                    0   
ENSG00000284332.1                     0                    0   

                   SU2CLC-CLE-NIVO20-T1  SU2CLC-CLE-NIVO21-T1  \
Name                                                            
ENSG00000223972.5                     1                     0   
ENSG00000227232.5                   203                  1730   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     0                     0   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO24-T1  SU2CLC-CLE-NIVO3-T1  \
Name                                                           
ENSG00000223972.5                     2                    5   
ENSG00000227232.5                    22                  437   
ENSG00000278267.1                     0                    0   
ENSG00000243485.5                     7                    0   
ENSG00000284332.1                     0                    0   

                   SU2CLC-CLE-NIVO31-T1  SU2CLC-CLE-NIVO47-T1  \
Name                                                            
ENSG00000223972.5                     0                     0   
ENSG00000227232.5                     3                     9   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     1                     1   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO5-T1  SU2CLC-CLE-NIVO52-T1  \
Name                                                           
ENSG00000223972.5                    0                     0   
ENSG00000227232.5                  117                     2   
ENSG00000278267.1                    0                     0   
ENSG00000243485.5                    0                     0   
ENSG00000284332.1                    0                     0   

                   SU2CLC-CLE-NIVO54-T1  SU2CLC-CLE-NIVO61-T1  \
Name                                                            
ENSG00000223972.5                     0                     0   
ENSG00000227232.5                   197                    79   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     0                     0   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO64-T1  SU2CLC-CLE-NIVO65-T1  \
Name                                                            
ENSG00000223972.5                     0                     0   
ENSG00000227232.5                     1                    96   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     0                     0   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO9-T1  SU2CLC-COL-1001-T1  \
Name                                                         
ENSG00000223972.5                    2                   4   
ENSG00000227232.5                  108                 269   
ENSG00000278267.1                    0               

The above matrix is organized in the following way:
- for each row: Name of gene
- for each column: {Descriprion: Name of patient, SU2CLC-CLE-NIVO10-T1:Participant_ID with type of therapy (T1)}

The idea is to organize the matrix in a better way for future analysis, in particular:
- for each row: Name of patients
- for each column: Name of gene

At this point we make an intersection between the columns of RNA data and the column Harmonized_SU2C_RNA_Tumor_Sample_ID_v2 of Clinical data.

In [ ]:
## Check
master_annotations_df["Harmonized_SU2C_RNA_Tumor_Sample_ID_v2"]

0                       NaN
1      SU2CLC-CLE-NIVO10-T1
2      SU2CLC-CLE-NIVO18-T1
3      SU2CLC-CLE-NIVO19-T1
4       SU2CLC-CLE-NIVO2-T1
               ...         
388                     NaN
389                     NaN
390                     NaN
391                     NaN
392                     NaN
Name: Harmonized_SU2C_RNA_Tumor_Sample_ID_v2, Length: 393, dtype: object

In [ ]:
## Take common patients that will be used in RNA analysis
common_patients = list(set(su2c_rna_counts_mat.columns.values).intersection(master_annotations_df["Harmonized_SU2C_RNA_Tumor_Sample_ID_v2"]))
len(common_patients)

# Save the list to a text file
with open(save_path_features + "common_patients.txt", "w") as file:
    for item in common_patients:
        file.write(str(item) + "\n")

## The idea is to reduce the master annotation data using this set of common genes and make the EDA using only this subset

In [ ]:
## Save description
description_rna_gene = su2c_rna_counts_mat["Description"]

## FIlter by QC
rna_master_annotations_df = master_annotations_df[(master_annotations_df["Pre-treatment_RNA_Sample_QC"] == "Keep") | (master_annotations_df["Pre-treatment_RNA_Sample_QC"] == "Flag")]
all_rna_samples = rna_master_annotations_df["Harmonized_SU2C_RNA_Tumor_Sample_ID_v2"].dropna().astype(str)

## FIltering and select index
rna_counts_mat_hq = su2c_rna_counts_mat[all_rna_samples]
rna_counts_mat_hq = rna_counts_mat_hq.set_index(su2c_rna_counts_mat.index)

In [ ]:
## Check if patients are unique
rna_counts_mat_hq.index.is_unique

True

In [ ]:
## Check Master Annotation Clinical Data with highest quality (n = 152)
print("Shape is: ", rna_master_annotations_df.shape)
display(rna_master_annotations_df.head())

Shape is:  (152, 38)


WES_Cohort_1  WES_Cohort_2  WES_All  RNA_Cohort_1  RNA_Cohort_2  RNA_All  \
1           NaN           NaN      NaN           1.0           NaN      1.0   
2           1.0           NaN      1.0           1.0           NaN      1.0   
3           1.0           NaN      1.0           1.0           NaN      1.0   
4           NaN           NaN      NaN           1.0           NaN      1.0   
5           NaN           NaN      NaN           1.0           NaN      1.0   

        Institution Harmonized_SU2C_Participant_ID_v2  \
1  Cleveland Clinic                 SU2CLC-CLE-NIVO10   
2  Cleveland Clinic                 SU2CLC-CLE-NIVO18   
3  Cleveland Clinic                 SU2CLC-CLE-NIVO19   
4  Cleveland Clinic                  SU2CLC-CLE-NIVO2   
5  Cleveland Clinic                 SU2CLC-CLE-NIVO20   

  Harmonized_SU2C_WES_Tumor_Sample_ID_v2  \
1                                    NaN   
2                   SU2CLC-CLE-NIVO18-T1   
3                   SU2CLC-CLE-NIVO19-T1   
4                                    NaN   
5                                    NaN   

  Harmonized_SU2C_WES_Normal_Sample_ID_v2  \
1                                     NaN   
2                    SU2CLC-CLE-NIVO18-N1   
3                    SU2CLC-CLE-NIVO19-N1   
4                                     NaN   
5                                     NaN   

  Harmonized_SU2C_RNA_Tumor_Sample_ID_v2 Pre-treatment_RNA_Sample_QC  \
1                   SU2CLC-CLE-NIVO10-T1                        Flag   
2                   SU2CLC-CLE-NIVO18-T1                        Keep   
3                   SU2CLC-CLE-NIVO19-T1                        Keep   
4                    SU2CLC-CLE-NIVO2-T1                        Keep   
5                   SU2CLC-CLE-NIVO20-T1                        Keep   

  Patient_Age_at_Diagnosis Patient_Sex  Patient_Race  Patient_Smoking_Status  \
1                       55           F           2.0                     1.0   
2                       68           F           0.0                     0.0   
3                       57           F           2.0                     2.0   
4                       63           F           0.0                     0.0   
5                       63           F           0.0                     1.0   

   Patient_Smoking_Pack_Years_Harmonized Histology_Harmonized  \
1                                   30.0                Adeno   
2                                    0.0                Adeno   
3                                   15.0                Adeno   
4                                    0.0                Adeno   
5                                   35.0                Adeno   

  Histology_Detail  Initial_Stage Initial_Stage_Substage  PDL1_TPS  \
1              NaN            4.0                    NaN       NaN   
2              NaN            2.0                      A       NaN   
3              NaN            4.0                    NaN       NaN   
4              NaN            4.0                    NaN       NaN   
5              NaN            4.0                    NaN       NaN   

  PDL1_TPS_Description Local_Antibody_Clone Clinical_Driver  \
1                  NaN                  NaN             NaN   
2                  NaN                  NaN            EGFR   
3                  NaN                  NaN             NaN   
4                  NaN                  NaN             NaN   
5                  NaN                  NaN            KRAS   

  Sequencing_Platform  Advanced_Diagnosis_Date  Line_of_Therapy  Agent_PD1  \
1                 NaN                   -321.0              3.0  Nivolumab   
2                 NaN                   -533.0              4.0  Nivolumab   
3                 NaN                    -35.0              1.0  Nivolumab   
4                 NaN                   -262.0              4.0  Nivolumab   
5                 NaN                   -301.0              2.0  Nivolumab   

  Agent_PD1_Category  Prior_Platinum  Prior_TKI  Harmonized_PFS_Event  \
1 

In [ ]:
## Check RNA Sample Counts with Highest Quality (n = 152)
print("Shape is: ", rna_counts_mat_hq.shape)
display(rna_counts_mat_hq.head())

Shape is:  (57523, 152)


SU2CLC-CLE-NIVO10-T1  SU2CLC-CLE-NIVO18-T1  \
Name                                                            
ENSG00000223972.5                     0                     0   
ENSG00000227232.5                    95                    96   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     0                     0   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO19-T1  SU2CLC-CLE-NIVO2-T1  \
Name                                                           
ENSG00000223972.5                    23                   36   
ENSG00000227232.5                   663                  395   
ENSG00000278267.1                     0                    0   
ENSG00000243485.5                     0                    0   
ENSG00000284332.1                     0                    0   

                   SU2CLC-CLE-NIVO20-T1  SU2CLC-CLE-NIVO21-T1  \
Name                                                            
ENSG00000223972.5                     1                     0   
ENSG00000227232.5                   203                  1730   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     0                     0   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO24-T1  SU2CLC-CLE-NIVO3-T1  \
Name                                                           
ENSG00000223972.5                     2                    5   
ENSG00000227232.5                    22                  437   
ENSG00000278267.1                     0                    0   
ENSG00000243485.5                     7                    0   
ENSG00000284332.1                     0                    0   

                   SU2CLC-CLE-NIVO31-T1  SU2CLC-CLE-NIVO47-T1  \
Name                                                            
ENSG00000223972.5                     0                     0   
ENSG00000227232.5                     3                     9   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     1                     1   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO5-T1  SU2CLC-CLE-NIVO52-T1  \
Name                                                           
ENSG00000223972.5                    0                     0   
ENSG00000227232.5                  117                     2   
ENSG00000278267.1                    0                     0   
ENSG00000243485.5                    0                     0   
ENSG00000284332.1                    0                     0   

                   SU2CLC-CLE-NIVO54-T1  SU2CLC-CLE-NIVO61-T1  \
Name                                                            
ENSG00000223972.5                     0                     0   
ENSG00000227232.5                   197                    79   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     0                     0   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO64-T1  SU2CLC-CLE-NIVO65-T1  \
Name                                                            
ENSG00000223972.5                     0                     0   
ENSG00000227232.5                     1                    96   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     0                     0   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO9-T1  SU2CLC-COL-1001-T1  \
Name                                                         
ENSG00000223972.5                    2                   4   
ENSG00000227232.5                  108                 269   
ENSG00000278267.1                    0                   0   
ENSG00000243485.5                    0                   0   
ENSG00000284332.1   

In [ ]:
## Add Description column
rna_counts_mat_hq.insert(0, "Description", description_rna_gene)

In [ ]:
## Check
print("Shape is: ", rna_counts_mat_hq.shape)
display(rna_counts_mat_hq.head())

Shape is:  (57523, 153)


Description  SU2CLC-CLE-NIVO10-T1  SU2CLC-CLE-NIVO18-T1  \
Name                                                                         
ENSG00000223972.5      DDX11L1                     0                     0   
ENSG00000227232.5       WASH7P                    95                    96   
ENSG00000278267.1    MIR6859-1                     0                     0   
ENSG00000243485.5  MIR1302-2HG                     0                     0   
ENSG00000284332.1    MIR1302-2                     0                     0   

                   SU2CLC-CLE-NIVO19-T1  SU2CLC-CLE-NIVO2-T1  \
Name                                                           
ENSG00000223972.5                    23                   36   
ENSG00000227232.5                   663                  395   
ENSG00000278267.1                     0                    0   
ENSG00000243485.5                     0                    0   
ENSG00000284332.1                     0                    0   

                   SU2CLC-CLE-NIVO20-T1  SU2CLC-CLE-NIVO21-T1  \
Name                                                            
ENSG00000223972.5                     1                     0   
ENSG00000227232.5                   203                  1730   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     0                     0   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO24-T1  SU2CLC-CLE-NIVO3-T1  \
Name                                                           
ENSG00000223972.5                     2                    5   
ENSG00000227232.5                    22                  437   
ENSG00000278267.1                     0                    0   
ENSG00000243485.5                     7                    0   
ENSG00000284332.1                     0                    0   

                   SU2CLC-CLE-NIVO31-T1  SU2CLC-CLE-NIVO47-T1  \
Name                                                            
ENSG00000223972.5                     0                     0   
ENSG00000227232.5                     3                     9   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     1                     1   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO5-T1  SU2CLC-CLE-NIVO52-T1  \
Name                                                           
ENSG00000223972.5                    0                     0   
ENSG00000227232.5                  117                     2   
ENSG00000278267.1                    0                     0   
ENSG00000243485.5                    0                     0   
ENSG00000284332.1                    0                     0   

                   SU2CLC-CLE-NIVO54-T1  SU2CLC-CLE-NIVO61-T1  \
Name                                                            
ENSG00000223972.5                     0                     0   
ENSG00000227232.5                   197                    79   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     0                     0   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO64-T1  SU2CLC-CLE-NIVO65-T1  \
Name                                                            
ENSG00000223972.5                     0                     0   
ENSG00000227232.5                     1                    96   
ENSG00000278267.1                     0                     0   
ENSG00000243485.5                     0                     0   
ENSG00000284332.1                     0                     0   

                   SU2CLC-CLE-NIVO9-T1  SU2CLC-COL-1001-T1  \
Name                                                         
ENSG00000223972.5                    2                   4   
ENSG00000227232.5                  108                 269   
ENSG00000278267.1                    0               

In [ ]:
## Check if all patients are in common with rna_counts_mat_hq and common_patients
all_present = all(item in rna_counts_mat_hq.columns.values for item in common_patients)
if all_present:
    print("All values in list1 are present in list2")
else:
    print("Not all values in list1 are present in list2")

All values in list1 are present in list2


In [ ]:
## Save the whole dataset without Description column (n = 152)
with open(save_path_data + "rna_counts.csv", "w") as output:
    output.write(str(rna_counts_mat_hq.to_csv(index = True)))

In [ ]:
## Save the whole dataset with Description column (n = 152)
with open(save_path_data + "rna_counts_with_desc.csv", "w") as output:
    output.write(str(su2c_rna_counts_mat.to_csv()))

In [ ]:
## Save the whole clinical dataset with with highest quality (n = 152)
with open(save_path_data + "master_annotations_df_hq.csv", "w") as output:
    output.write(str(rna_master_annotations_df.to_csv()))

### Read WES (Exome) Data

In [ ]:
## Load Data
panlung_drivers = pd.read_csv(source_path_data_ref + "panlung_drivers.txt", sep = "\t", header = None).iloc[:, 0].to_list()

In [ ]:
## Load wes data
su2c_wes_data = pd.read_csv(source_path_data_exome + "SU2C-MARK_Harmonized_Validated_Damaging_Pivot_Bin_v1.txt", sep = "\t")

## Take only some columns
su2c_wes_data_drivers = su2c_wes_data.loc[:, su2c_wes_data.columns.isin(panlung_drivers)]
su2c_wes_data_drivers["Tumor_Sample_Barcode"] = su2c_wes_data["Tumor_Sample_Barcode"]
## Set index
su2c_wes_data_drivers = su2c_wes_data_drivers.set_index("Tumor_Sample_Barcode")

In [ ]:
## Check
print("Shape is: ", su2c_wes_data_drivers.shape)
display(su2c_wes_data_drivers.head())

Shape is:  (309, 52)


APC  ARHGAP35  ARID1A  ATF7IP  ATM  B2M  BRAF  CDKN2A  \
Tumor_Sample_Barcode                                                          
SU2CLC-CLE-NIVO1-T1     0         0       0       0    0    0     0       0   
SU2CLC-CLE-NIVO18-T1    0         0       0       0    0    0     0       0   
SU2CLC-CLE-NIVO19-T1    0         0       1       0    0    0     1       0   
SU2CLC-CLE-NIVO3-T1     0         0       0       0    1    0     0       0   
SU2CLC-CLE-NIVO4-T1     0         0       0       0    0    0     0       0   

                      CMTR2  COL5A2  CREBBP  CTNNB1  CUL3  DSN1  EGFR  ELL2  \
Tumor_Sample_Barcode                                                          
SU2CLC-CLE-NIVO1-T1       0       0       0       0     0     0     0     0   
SU2CLC-CLE-NIVO18-T1      0       0       0       0     0     0     1     0   
SU2CLC-CLE-NIVO19-T1      0       0       1       0     0     0     0     0   
SU2CLC-CLE-NIVO3-T1       0       0       0       0     0     0     0     0   
SU2CLC-CLE-NIVO4-T1       0       0       0       0     0     0     0     0   

                      EP300  ERBB2  FANCM  FAT1  FBXW7  HLA-A  HRAS  ITGBL1  \
Tumor_Sample_Barcode                                                          
SU2CLC-CLE-NIVO1-T1       0      0      0     0      0      0     0       0   
SU2CLC-CLE-NIVO18-T1      0      0      0     0      0      0     0       0   
SU2CLC-CLE-NIVO19-T1      0      0      0     0      0      0     0       0   
SU2CLC-CLE-NIVO3-T1       0      0      0     0      0      0     0       0   
SU2CLC-CLE-NIVO4-T1       0      0      0     0      0      0     0       0   

                      KEAP1  KLF5  KLHL5  KMT2C  KMT2D  KRAS  LATS1  MET  MGA  \
Tumor_Sample_Barcode                                                            
SU2CLC-CLE-NIVO1-T1       0     0      0      0      1     0      0    0    1   
SU2CLC-CLE-NIVO18-T1      0     0      0      0      0     0      0    0    0   
SU2CLC-CLE-NIVO19-T1      0     0      0      0      0     0      0    0    0   
SU2CLC-CLE-NIVO3-T1       0     0      0      0      0     0      0    0    0   
SU2CLC-CLE-NIVO4-T1       0     0      0      0      0     0      0    0    1   

                      NCOA6  NF1  NFE2L2  NOTCH1  NRAS  PIK3CA  PLXNB2  PTEN  \
Tumor_Sample_Barcode                                                           
SU2CLC-CLE-NIVO1-T1       0    0       0       0     0       0       0     0   
SU2CLC-CLE-NIVO18-T1      0    0       0       0     0       0       0     0   
SU2CLC-CLE-NIVO19-T1      0    0       0       0     0       0       0     0   
SU2CLC-CLE-NIVO3-T1       0    0       0       0     0       0       0     0   
SU2CLC-CLE-NIVO4-T1       0    0       0       0     0       0       0     0   

                      RASA1  RB1  RBM10  RIT1  SETD2  SMAD4  SMARCA4  STK11  \
Tumor_Sample_Barcode                                                          
SU2CLC-CLE-NIVO1-T1       0    0      0     0      0      0        0      0   
SU2CLC-CLE-NIVO18-T1      0    0      0     0      0      0        0      0   
SU2CLC-CLE-NIVO19-T1      0    0      0     0      0      0        0      0   
SU2CLC-CLE-NIVO3-T1       0    0      0     0      0      0        0      0   
SU2CLC-CLE-NIVO4-T1       0    0      0     0      0      0        0      0   

                      TP53  U2AF1  ZFP36L1  
Tumor_Sample_Barcode                        
SU2CLC-CLE-NIVO1-T1      1      0        0  
SU2CLC-CLE-NIVO18-T1     0      0        0  
SU2CLC-CLE-NIVO19-T1     0      0        0  
SU2CLC-CLE-NIVO3-T1      1      0        0  
SU2CLC-CLE-NIVO4-T1      1      0        0

In [ ]:
## Save the whole dataset
with open(save_path_data + "wes_counts.csv", "w") as output:
    output.write(str(su2c_wes_data_drivers.to_csv()))

So, the files in which we are interessed are in **save_path_data**:
- **rna_counts.csv** for *RNA-seq data*.
- **wes_counts.csv** for *WES data*.
- **master_annotation.csv** for *Clinical data.*


For further analysis it's important to understand if there are some genes in WES data that are equal to RNA-seq data.

In [ ]:
## Number of genes for WES data
description_wes_genes = su2c_wes_data_drivers.columns.values.tolist()
print("Number of genes for WES data: ", len(description_wes_genes))
## Number of genes for RNA-seq data
print("Number of genes for RNA-seq data: ", len(description_rna_gene))
print("\n")

## So, there are at most 52 genes in common between the two datasets.

## Check if the columns in Exome data (genes) are the same (also in part) of Gene in RNA-seq
common_genes = list(set(description_rna_gene).intersection(description_wes_genes))
print("Number of genes in common: ", len(common_genes))
print("Genes: ", common_genes)

Number of genes for WES data:  52
Number of genes for RNA-seq data:  57523


Number of genes in common:  52
Genes:  ['HLA-A', 'EGFR', 'PTEN', 'NF1', 'BRAF', 'RBM10', 'ELL2', 'HRAS', 'COL5A2', 'CMTR2', 'KLF5', 'FAT1', 'KRAS', 'MGA', 'RIT1', 'ATM', 'PLXNB2', 'ZFP36L1', 'EP300', 'NCOA6', 'U2AF1', 'CDKN2A', 'RASA1', 'MET', 'RB1', 'ITGBL1', 'KMT2C', 'SMARCA4', 'PIK3CA', 'KEAP1', 'ERBB2', 'ARID1A', 'NRAS', 'CREBBP', 'TP53', 'DSN1', 'FANCM', 'STK11', 'ATF7IP', 'CTNNB1', 'FBXW7', 'NOTCH1', 'SMAD4', 'SETD2', 'NFE2L2', 'CUL3', 'APC', 'KMT2D', 'ARHGAP35', 'B2M', 'KLHL5', 'LATS1']


In [ ]:
## Check 2
## Define a function to check if a gene is present in a list
def present_or_not(list_genes, gene):
    for i in range(len(gene)):
        if gene[i] in list_genes.values:
            print(f"{gene[i]} present")
        else:
            print(f"{gene[i]} not present")

In [ ]:
## Apply
present_or_not(description_rna_gene, common_genes)

HLA-A present
EGFR present
PTEN present
NF1 present
BRAF present
RBM10 present
ELL2 present
HRAS present
COL5A2 present
CMTR2 present
KLF5 present
FAT1 present
KRAS present
MGA present
RIT1 present
ATM present
PLXNB2 present
ZFP36L1 present
EP300 present
NCOA6 present
U2AF1 present
CDKN2A present
RASA1 present
MET present
RB1 present
ITGBL1 present
KMT2C present
SMARCA4 present
PIK3CA present
KEAP1 present
ERBB2 present
ARID1A present
NRAS present
CREBBP present
TP53 present
DSN1 present
FANCM present
STK11 present
ATF7IP present
CTNNB1 present
FBXW7 present
NOTCH1 present
SMAD4 present
SETD2 present
NFE2L2 present
CUL3 present
APC present
KMT2D present
ARHGAP35 present
B2M present
KLHL5 present
LATS1 present
